internal/operator-controller/rukpak/util/util.go

```diff
@@ -17,6 +19,21 @@ func ObjectNameForBaseAndSuffix(base string, suffix string) string {
 	return fmt.Sprintf("%s-%s", base, suffix)
 }
 
+func ToUnstructured(obj client.Object) (*unstructured.Unstructured, error) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
```

Comment:
A `client.Object` is not guaranteed to have a populated GVK. If we need guarantees about setting the GVK on the unstructured type, we should also pass in a scheme so that we can lookup the GVK based on the object type.

Reply:
See https://github.com/kubernetes-sigs/controller-runtime/blob/6ad5c1dd4418489606d19dfb87bf38905b440561/pkg/client/apiutil/apimachinery.go#L95

Reply:
For now, I've just hardened it by checking if version and kind are empty. I don't think we need to add the complexity of passing a scheme. At least not for now =D

Reply:
If we want the GKV and use it is the right approach then, I think we should use the controller-runtime implementation to do so: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/client/apiutil/apimachinery.go#L95C6-L95C18

Reply:
I don't think we need it right now. The precondition for the function is that the client.Object will carry the GVK. If not, then error.

Reply:
Do you think that's acceptable since we only use it with CertManager? That way, we'll have the GKV. **Is that what you were thinking?**

Reply:
Yeah, if we check/fail if GVK is unset, then I think that's sufficient.